clustered/client/src/main/java/org/ehcache/clustered/config/xml/ClusteringServiceConfigurationParser.java

```diff
@@ -0,0 +1,104 @@
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.config.xml;
+
+import org.ehcache.clustered.config.ClusteringServiceConfiguration;
+import org.ehcache.clustered.service.ClusteringService;
+import org.ehcache.spi.service.ServiceCreationConfiguration;
+import org.ehcache.xml.CacheManagerServiceConfigurationParser;
+import org.w3c.dom.Attr;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.io.IOException;
+import java.net.URI;
+
+import javax.xml.transform.Source;
+import javax.xml.transform.stream.StreamSource;
+
+import static org.ehcache.clustered.config.xml.ClusteredCacheConstants.*;
+
+/**
+ * Provides parsing support for the {@code <service>} elements representing a {@link ClusteringService ClusteringService}.
+ *
+ * @author Clifford W. Johnson
+ *
+ * @see ClusteredCacheConstants#XSD
+ */
+public class ClusteringServiceConfigurationParser implements CacheManagerServiceConfigurationParser<ClusteringService> {
+
+  @Override
+  public Source getXmlSchema() throws IOException {
+    return new StreamSource(XML_SCHEMA.openStream());
+  }
+
+  @Override
+  public URI getNamespace() {
+    return NAMESPACE;
+  }
+
+  /**
+   * Complete interpretation of the top-level elements defined in <code>{@value ClusteredCacheConstants#XSD}</code>.
+   * This method is called only for those elements from the namespace set by {@link ClusteredCacheConstants#NAMESPACE}.
+   *
+   * @param fragment the XML fragment to process
+   *
+   * @return a {@link org.ehcache.clustered.config.ClusteringServiceConfiguration ClusteringServiceConfiguration}
+   */
+  @Override
+  public ServiceCreationConfiguration<ClusteringService> parseServiceCreationConfiguration(final Element fragment) {
+    /*
+     * The presumptions in this code are:
+     *   1) the Element is schema-validated
+     *   2) 'connection' is defined as required (minOccurs=1, maxOccurs=1)
+     *   3) 'connection.url' is defined as "required"
+     *   4) 'cluster.alias' is optional
+     *   5) no other elements or attributes are defined
+     */
+    if ("cluster".equals(fragment.getLocalName())) {
+
+      final Attr aliasAttribute = fragment.getAttributeNode("alias");
+      final String alias = (aliasAttribute == null ? "" : aliasAttribute.getValue());
+
+      URI connectionUri = null;
+      final NodeList childNodes = fragment.getChildNodes();
+      for (int i = 0; i < childNodes.getLength(); i++) {
+        final Node item = childNodes.item(i);
+        if (Node.ELEMENT_NODE == item.getNodeType()) {
+          if ("connection".equals(item.getLocalName())) {
+            final Attr urlAttribute = ((Element)item).getAttributeNode("url");
+            final String urlValue = urlAttribute.getValue();
+            try {
+              connectionUri = URI.create(urlValue);
+            } catch (IllegalArgumentException e) {
+              throw new RuntimeException(
```

Comment:
I've added an `XmlConfigurationException` to the **xml** module ... looks like there are a number of places where it might be brought to bear.  Should I create an issue to cover that work?